docs/lib/project-setup/fragments/flutter/null-safety/null-safety.md

```diff
@@ -0,0 +1,20 @@
+**Amplify Flutter and Null Safety**
+
+
+The Amplify Flutter library supports [Dart null safety](https://dart.dev/null-safety) beginning with version 0.2.0.  This means that:
+
+
+
+* Applications utilizing [sound null safety](https://dart.dev/null-safety) must use Amplify Flutter version 0.2.0 or greater.
+
+* Applications utilizing [unsound null safety](https://dart.dev/null-safety/unsound-null-safety) can use versions of Amplify Flutter above or below 0.2.0, but some application changes may be necessary to migrate to 0.2.0.
+
+* Applications that are not using Flutter 2.0 or higher must use a version of Amplify Flutter lower than 0.2.0.
+
+
+**Using Generated Codegen Models with Null Safety**
+
+
+The Amplify CLI's `codegen models` command can generate Dart models with or without null safety. To generate null safe models, make sure that the `pubspec.yaml` file at the root of your projects defines a Dart SDK version of 2.12.0 or greater, and make sure that the `enableDartNullSafety` [feature flag](https://docs.amplify.aws/cli/reference/feature-flags) is set to true in your `amplify/cli.json` file.
+
+In order to migrate to Amplify Flutter version 0.2.0 or greater, you will need to regenerate your codegen models with null safety (using the `amplify codegen models` command) if you are using them.
```

Comment:
I think we should include a migration path to null safety for apps using datastore. For all other packages, the upgrade path is no different that any other package (upgrade all dependencies, then upgrade the app). Data store has a third step of generating the new models.

The steps aren't really that much different, but I think we should just be specific:
1. Confirm your app is working with the latest non null safe version of amplify-flutter
2. Upgrade amplify-flutter to 0.2.0 and make updates for breaking changes (see package change logs for details).
3. Upgrade app to null safety: For the application code follow flutter docs, for the model code re-generate the models

Open question if steps 2 and 3 should combined. I believe the app should work in the mixed state (non null safe app, non null safe models, null safe amplify-flutter). If it doesn't, they should probably be listed as one step w/ a specific callout that both steps need to be done together.

Reply:
I think the open question has been answered. Steps 2 and 3 can be distinct steps and the application should function as expected after each step. With that in mind, rather than listing out the migration steps as I suggested above, I think we could probably just update this line to be:

> If your app uses generated models from amplify, you will need to migrate these models to be null safe as part of migrating your app to be null safe. You should follow the [flutter null safety migration docs](https://dart.dev/null-safety/migration-guide) to migrate your application code, excluding the generated models. To migrate the models to be null safe, you can simply regenerate them following the instructions above.

I think as is, the line is not correct. You do not have to regenerate models to use 0.2.0. You just have to regenerate models to migrate your app to null safety.